crates/recursor/src/recursor.rs

```diff
@@ -487,7 +487,22 @@ mod for_dnssec {
 
                         DnsResponse::new(msg, vec![])
                     })
-                    .map_err(|e| ProtoError::from(e.to_string()))
+                    .map_err(|e| {
+                        // This is a very incomplete mapping, and should be an Into
+                        // implementation...
```

Comment:
Sounds like something that might best be addressed in this PR?

Reply:
Once I get a handle on the overall scope of everything that needs to change to fix these problems, I'll clean these interfaces up/complete them.